packages/cli/src/commands/integration/add.ts

```diff
@@ -1,116 +1,154 @@
   if (args.length > 1) {
-    client.output.error(`Can't install more than one integration at a time.`);
+    client.output.error('Cannot install more than one integration at a time');
     return 1;
   }
 
   const integrationSlug = args[0];
 
-  const user = await getUser(client);
-  const teamId = client.config.currentTeam ?? user.defaultTeamId;
+  if (!integrationSlug) {
+    client.output.error('You must pass an integration slug');
+    return 1;
+  }
+
+  const { contextName, team } = await getScope(client);
 
-  if (!teamId) {
+  if (!team) {
     client.output.error('Team not found');
     return 1;
   }
 
-  const team = await getTeamById(client, teamId);
-  const integration = await fetchIntegration(client, team, integrationSlug);
-
-  if (!integration) {
-    client.output.error(`Integration not found: ${integrationSlug}`);
+  let integration: Integration | undefined;
+  try {
+    integration = await fetchIntegration(client, integrationSlug);
+  } catch (error) {
+    client.output.error(
+      `Failed to get integration "${integrationSlug}": ${(error as Error).message}`
+    );
     return 1;
   }
 
   if (!integration.products) {
     client.output.error(
-      `Integration is not from the marketplace: ${integration.name}`
+      `Integration "${integrationSlug}" is not a Marketplace integration`
     );
     return 1;
   }
 
-  const product = await selectProduct(client, integration);
+  const [productResult, installationsResult] = await Promise.allSettled([
+    selectProduct(client, integration),
+    fetchInstallations(client, integration),
+  ]);
 
-  if (!product) {
+  if (productResult.status === 'rejected' || !productResult.value) {
+    client.output.error('Product not found');
+    return 1;
+  }
+
+  if (installationsResult.status === 'rejected') {
     client.output.error(
-      `No products found for integration: ${integration.name}`
+      `Failed to get integration installations: ${installationsResult.reason}`
     );
     return 1;
   }
 
-  const projectLink = await getOptionalLinkedProject(client);
+  const product = productResult.value;
+  const installations = installationsResult.value;
 
-  if (projectLink?.status === 'error') {
-    return projectLink.exitCode;
+  const teamInstallations = installations.filter(
+    install =>
+      install.ownerId === team.id && install.installationType === 'marketplace'
+  );
+
+  if (teamInstallations.length > 1) {
+    client.output.error(
+      `Found more than one existing installation of ${integration.name}. Please contact Vercel Support at https://vercel.com/help`
+    );
+    return 1;
   }
 
-  privisionResourceViaWebUI(
-    client,
-    teamId,
-    integration.id,
-    product.id,
-    projectLink?.project?.id
+  const installation = teamInstallations[0] as
+    | IntegrationInstallation
+    | undefined;
+
+  client.output.log(
+    `Installing ${chalk.bold(product.name)} by ${chalk.bold(integration.name)} under ${chalk.bold(contextName)}`
   );
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏consider adding spinner before this

I just tried this locally:

```
vc --cwd=/Users/smassa/source/demo/nextjs i supabase
```

And waited over 10 seconds to see any output. Will this be common? We may want a spinner here.

Reply:
Interesting, it has been fast for me so far. We'll have to monitor that. I'd rather make the APIs faster, but there are some provider APIs that might be slow. I'll check that and might do a follow up PR.